src/gordon_gcp/clients/http.py

```diff
@@ -142,42 +142,50 @@ def __init__(self, auth_client=None, session=None):
             method=method.upper(),
             url=url,
             kwargs=req_kwargs))
-        async with self._session.request(
-                method, url, **req_kwargs) as resp:
-            log_kw = {
-                'request_id': request_id,
-                'method': method.upper(),
-                'url': resp.url,
-                'status': resp.status,
-                'reason': resp.reason
-            }
-            logging.debug(_utils.RESP_LOG_FMT.format(**log_kw))
-
-            if resp.status in REFRESH_STATUS_CODES:
-                logging.warning(f'[{request_id}] HTTP Status Code {resp.status}'
-                                f' returned requesting {resp.url}: '
-                                f'{resp.reason}')
-                if token_refresh_attempts:
-                    logging.info(f'[{request_id}] Attempting request to '
-                                 f'{resp.url} again.')
-                    return await self.request(
-                        method, url,
-                        token_refresh_attempts=token_refresh_attempts,
-                        request_id=request_id,
-                        **req_kwargs)
-
-                logging.warning(f'[{request_id}] Max attempts refreshing auth '
-                                f'token exhausted while requesting {resp.url}')
-
-            # avoid leaky abstractions and wrap http errors with our own
-            try:
-                resp.raise_for_status()
-            except aiohttp.ClientResponseError as e:
-                msg = f'[{request_id}] Issue connecting to {resp.url}: {e}'
-                logging.error(msg, exc_info=e)
-                raise exceptions.GCPHTTPResponseError(msg, resp.status)
-
-            return await resp.text()
+        try:
+            resp = await self._session.request(method, url, **req_kwargs)
+        except Exception as e:
+            msg = f'[{request_id}] Request call failed {resp.url}: {e}'
+            logging.error(msg, exc_info=e)
+            resp.release()
+            raise e
```

Comment:
Since we wrap aiohttp exceptions in our own (i.e. below in line 184 of the patch), I would want us to raise our own exception (i.e. `exceptions.GCPHTTPResponseError`) . Otherwise, the code calling this code will have to be updated to catch this mystery exception.

Reply:
Also I just realized, the `resp` variable will not be available to you for the `msg` string nor `release` it since that's what raised the exception. That makes me wonder what you exactly you see when you're testing this locally, i.e. are you ever able to get to this point.

Reply:
Thaaaaat... is a good point.  I tested it by raising an exception after that line.  The good news is we don't need it for the msg as long as we have the request ID.